### ISSUE TITLE
git clone private repo got could not read username

### ISSUE THREAD

**Issue by tingShean** [tektoncd/pipeline#3861]
# Expected Behavior
git clone private repo use basic-auth

# Actual Behavior
could not read Username 

# Steps to Reproduce the Problem

secret.yaml
```
apiVersion: v1
kind: Secret
metadata:
  name: git-sec
  namespace: tekton-task-git
  annotations:
    tekton.dev/git-0: https://private.git.site
type: "kubernetes.io/basic-auth"
stringData:
  username: myaccount
  password: {mypassword}
```
service_account.yaml
```
apiVersion: v1
kind: ServiceAccount
metadata:
  name: devops-git-sec
  namespace: tekton-task-git
secrets:
  - name: git-sec
```
taskrun.yaml
```
apiVersion: tekton.dev/v1beta1
kind: TaskRun
metadata:
  name: git-private
spec:
  serviceAccountName: devops-git-sec
  resources:
    inputs:
      - name: git-repo
        resourceSpec:
          type: git
          params:
            - name: revision
              value: master
            - name: url
              value: https://private.git.site/test/test.git
  taskSpec:
    resources:
      inputs:
      - name: git-repo
        type: git
    steps:
    - name: config
      image: ubuntu
      script: |
        #!/usr/bin/env bash
        cat /workspace/git-repo/README.md
```
Finally
```
{"level":"error","ts":1617177586.2977417,"caller":"git/git.go:54","msg":"Error running git [fetch --recurse-submodules=yes --depth=1 origin --update-head-ok --force master]: exit status 128\nfatal: could not read Username for 'https://private.git.site': No such device or address\n","stacktrace":"github.com/tektoncd/pipeline/pkg/git.run\n\tgithub.com/tektoncd/pipeline/pkg/git/git.go:54\ngithub.com/tektoncd/pipeline/pkg/git.Fetch\n\tgithub.com/tektoncd/pipeline/pkg/git/git.go:149\nmain.main\n\tgithub.com/tektoncd/pipeline/cmd/git-init/main.go:53\nruntime.main\n\truntime/proc.go:204"}
{"level":"fatal","ts":1617177586.2978184,"caller":"git-init/main.go:54","msg":"Error fetching git repository: failed to fetch [master]: exit status 128","stacktrace":"main.main\n\tgithub.com/tektoncd/pipeline/cmd/git-init/main.go:54\nruntime.main\n\truntime/proc.go:204"}
```

# Additional Info

- Kubernetes version:

  **Output of `kubectl version`:**

```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.6", GitCommit:"dff82dc0de47299ab66c83c626e08b245ab19037", GitTreeState:"clean", BuildDate:"2020-07-15T23:30:39Z", GoVersion:"go1.1
4.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"17+", GitVersion:"v1.17.12-eks-7684af", GitCommit:"7684af4ac41370dd109ac13817023cb8063e3d45", GitTreeState:"clean", BuildDate:"2020-10-20T22:57:40Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
```

- Tekton Pipeline version:

  **Output of `tkn version` or `kubectl get pods -n tekton-pipelines -l app=tekton-pipelines-controller -o=jsonpath='{.items[0].metadata.labels.version}'`**

```
dashboard v0.15.0
pipeline v0.22.0
```

**Comment by tingShean** [comment]
this issue maybe only in v0.22.0
change to use v0.21.0 work fine

**Comment by sbwsg** [comment]
Just wanted to follow up on this one - I wasn't able to reproduce with a private Github repo over https with a personal access token as the password.  This was running off the very tip of `main` so it might have been fixed in 0.23?  If you're still able to reproduce in later versions of Tekton please reopen this bug and we can investigate further.
